Library/Homebrew/cmd/update.sh

```diff
@@ -293,14 +295,18 @@ EOS
   # ensure GIT_CONFIG is unset as we need to operate on .git/config
   unset GIT_CONFIG
 
+  # only allow one instance of brew update
+  source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
```

Comment:
Like in Ruby code with `require`s, I'd prefer to move this `source` to the top of the file unless there's a very good reason for it to be here. (I don't think there is.)